### ISSUE TITLE
Is there any problem about the profile result?

### ISSUE THREAD

**Issue by Xreki** [PaddlePaddle/Paddle#15858]
After I update the code on the `develop` branch, I find the profiling result is different, see following:
```
------------------------->     Profiling Report     <-------------------------

Place: All
Time unit: ms
Sorted by total time in descending order in the same thread

Event                              Calls       Total       Min.        Max.        Ave.        Ratio.      
thread0::gru                       200         135.344     0.669756    0.765086    0.676719    0.352381    
thread0::mul                       6900        72.8319     0.002464    0.02592     0.0105553   0.189625    
thread0::sequence_expand           3900        43.3026     0.006655    0.02192     0.0111032   0.112743    
thread0::gru_unit                  1300        20.9425     0.015551    0.029088    0.0161096   0.054526    
thread0::conv2d                    800         19.1258     0.014496    0.040096    0.0239073   0.049796    
thread0::sequence_pool             1400        18.8459     0.003616    0.01504     0.0134614   0.0490672   
thread0::beam_search               1300        9.32532     0.00656     0.013568    0.00717332  0.0242794   
thread0::elementwise_add           5200        8.89447     0.001536    0.00576     0.00171047  0.0231577   
thread0::write_to_array            5500        7.3553      0           0.004288    0.00133733  0.0191503   
thread0::read_from_array           3900        5.56034     0.001311    0.004832    0.00142573  0.0144769   
thread0::beam_search_decode        100         5.24623     0.050399    0.06928     0.0524623   0.0136591   
thread0::top_k                     1300        4.76507     0.003488    0.005248    0.00366544  0.0124063   
thread0::batch_norm                800         4.57774     0.002432    0.01328     0.00572218  0.0119186   
thread0::sequence_softmax          1300        4.06955     0.003007    0.004512    0.00313043  0.0105955   
thread0::softmax                   1300        4.00113     0.003007    0.003392    0.0030778   0.0104174   
thread0::elementwise_mul           1300        3.56779     0.002624    0.003744    0.00274445  0.00928911  
thread0::lookup_table              1300        2.53189     0.001727    0.005696    0.00194761  0.00659204  
thread0::is_empty                  1300        2.41772     0           0.004032    0.00185978  0.00629477  
thread0::tanh                      1300        2.2735      0.001632    0.00368     0.00174884  0.00591928  
thread0::relu                      900         1.8422      0.001311    0.00384     0.00204689  0.00479636  
thread0::log                       1300        1.76604     0.001279    0.003776    0.00135849  0.00459807  
thread0::reshape2                  2600        1.67903     0           0.0032      0.000645782 0.00437154  
thread0::pool2d                    300         1.21968     0.0032      0.00544     0.00406558  0.00317555  
thread0::concat                    200         0.914014    0.004288    0.00704     0.00457007  0.00237973  
thread0::im2sequence               100         0.584127    0.005728    0.006016    0.00584127  0.00152083  
thread0::scale                     200         0.493757    0.001312    0.006432    0.00246879  0.00128555  
thread0::fetch                     200         0.445727    0.002048    0.004192    0.00222864  0.0011605   
thread0::cast                      200         0.160128    0           0.002912    0.00080064  0.00041691  
thread0::fill_constant             200         0           0           0           0           0           
thread0::increment                 1300        0           0           0           0           0           
thread0::lod_reset                 1300        0           0           0           0           0           
thread0::logical_not               1300        0           0           0           0           0           
thread0::logical_and               1300        0           0           0           0           0           
thread0::while                     100         0           0           0           0           0           
thread0::tensor_array_to_tensor    100         0           0           0           0           0           
thread0::less_than                 1400        0           0           0           0           0           
thread0::feed                      300         0           0           0           0           0
```

Profiling result before:
```
------------------------->     Profiling Report     <-------------------------

Place: All
Time unit: ms
Sorted by total time in descending order in the same thread

Event                              Calls       Total       Min.        Max.        Ave.        Ratio.      
thread0::while                     100         2299.59     16.5595     121.296     22.9959     0.456365    
thread0::gru                       200         565.844     2.00378     11.9398     2.82922     0.112295    
thread0::sequence_expand           3900        306.454     0.023264    103.035     0.078578    0.0608175   
thread0::mul                       6900        269.188     0.010912    3.39738     0.0390128   0.0534219   
thread0::gru_unit                  1300        163.871     0.086464    0.467488    0.126055    0.0325211   
thread0::beam_search               1300        152.938     0.0616      4.25187     0.117645    0.0303514   
thread0::elementwise_add           5200        152.303     0.009984    0.237632    0.029289    0.0302253   
thread0::write_to_array            5500        147.11      0.006176    0.292128    0.0267472   0.0291947   
thread0::read_from_array           3900        114.428     0.007968    2.59517     0.0293405   0.0227089   
thread0::beam_search_decode        100         91.8879     0.725376    3.47638     0.918879    0.0182357   
thread0::is_empty                  1300        84.8727     0.021632    4.90624     0.0652867   0.0168435   
thread0::conv2d                    800         73.8807     0.059584    0.830304    0.0923509   0.014662    
thread0::sequence_pool             1400        66.2023     0.01504     3.77773     0.0472874   0.0131382   
thread0::reshape2                  2600        63.0116     0.005056    0.120256    0.0242352   0.012505    
thread0::top_k                     1300        50.8271     0.012864    2.64282     0.0390978   0.0100869   
thread0::elementwise_mul           1300        48.5799     0.017184    0.2112      0.0373691   0.00964094  
thread0::softmax                   1300        46.1661     0.019584    0.205472    0.0355124   0.00916191  
thread0::lookup_table              1300        44.8221     0.013376    1.06662     0.0344786   0.0088952   
thread0::tanh                      1300        37.0956     0.012672    0.154368    0.0285351   0.00736182  
thread0::sequence_softmax          1300        34.1524     0.011392    0.110976    0.0262711   0.00677773  
thread0::log                       1300        34.0186     0.013984    0.216448    0.0261681   0.00675117  
thread0::batch_norm                800         25.6206     0.010944    0.122848    0.0320258   0.00508455  
thread0::relu                      900         21.5807     0.009856    0.71104     0.0239786   0.00428282  
thread0::less_than                 1400        19.6746     0.003104    0.095648    0.0140533   0.00390453  
thread0::increment                 1300        14.7496     0.003968    0.282592    0.0113458   0.00292714  
thread0::lod_reset                 1300        14.738      0.003104    0.065696    0.0113369   0.00292483  
thread0::logical_not               1300        14.0086     0.0056      0.047808    0.0107759   0.00278009  
thread0::logical_and               1300        13.2763     0.0048      0.056384    0.0102125   0.00263475  
thread0::pool2d                    300         12.8712     0.013888    2.22429     0.0429039   0.00255435  
thread0::tensor_array_to_tensor    100         12.5149     0.086176    0.325344    0.125149    0.00248365  
thread0::concat                    200         11.616      0.015552    0.529696    0.05808     0.00230526  
thread0::scale                     200         10.4477     0.017824    0.245792    0.0522384   0.0020734   
thread0::fetch                     200         8.60224     0.032224    0.13936     0.0430112   0.00170716  
thread0::cast                      200         4.65098     0.008896    0.078336    0.0232549   0.000923012 
thread0::im2sequence               100         3.61443     0.013632    0.375136    0.0361443   0.000717304 
thread0::feed                      300         1.86096     0.003072    0.126464    0.0062032   0.000369318 
thread0::fill_constant             200         1.84768     0.003104    0.03504     0.0092384   0.000366682
```

Now I am quite confusing ...

**Comment by qingqing01** [comment]
@Xreki Maybe related to https://github.com/PaddlePaddle/Paddle/pull/15301

**Comment by Xreki** [comment]
Discussed with @cjld and @qingqing01 , those ops' time is 0, because there is no `cuda` functions, including `cuda memcpy` and `cuda __global__` function. But other overhead, such as frameworks' overhead and CPU time cannot be shown and the profiling result is confusing for us. Maybe CPU time should be recorded and shown at the same time?

**Comment by Xreki** [comment]
Fixed by #15842 , now the profiling result is:
```
------------------------->     Profiling Report     <-------------------------

Place: All
Time unit: ms
Sorted by total time in descending order in the same thread

Event                              Calls       Total       CPU Time (Ratio)        GPU Time (Ratio)        Min.        Max.        Ave.        Ratio.      
thread0::while                     100         1424.49     1410.020740 (0.989840)  14.473088 (0.010160)    13.7816     20.3279     14.2449     0.380674    
thread0::gru                       200         615.933     486.353479 (0.789620)   129.579848 (0.210380)   2.85813     8.8623      3.07967     0.164599    
thread0::mul                       6900        219.691     147.190168 (0.669988)   72.500739 (0.330012)    0.018663    0.093212    0.0318393   0.0587091   
thread0::sequence_expand           3900        208.415     165.961472 (0.796303)   42.453533 (0.203697)    0.042877    0.201377    0.0534397   0.0556958   
thread0::gru_unit                  1300        152.785     131.974094 (0.863789)   20.811076 (0.136211)    0.104736    0.188825    0.117527    0.0408295   
thread0::elementwise_add           5200        137.322     128.567991 (0.936252)   8.754021 (0.063748)     0.022521    0.05165     0.0264081   0.0366972   
thread0::beam_search               1300        120.425     111.260987 (0.923899)   9.164480 (0.076101)     0.083777    0.171446    0.092635    0.0321819   
thread0::write_to_array            5500        109.266     102.148950 (0.934865)   7.117071 (0.065135)     0.004312    0.042772    0.0198665   0.0291997   
thread0::beam_search_decode        100         90.8835     85.174780 (0.937187)    5.708681 (0.062813)     0.822885    6.93055     0.908835    0.0242872   
thread0::read_from_array           3900        76.7875     71.461417 (0.930639)    5.326089 (0.069361)     0.015231    0.043538    0.0196891   0.0205203   
thread0::conv2d                    800         73.2853     54.148182 (0.738868)    19.137111 (0.261132)    0.070353    0.153523    0.0916066   0.0195844   
thread0::is_empty                  1300        61.2328     58.837103 (0.960875)    2.395712 (0.039125)     0.01878     0.073602    0.0471022   0.0163635   
thread0::sequence_pool             1400        47.4778     28.653668 (0.603518)    18.824102 (0.396482)    0.031384    0.070172    0.0339127   0.0126877   
thread0::reshape2                  2600        42.5929     40.914562 (0.960596)    1.678309 (0.039404)     0.006075    0.043813    0.0163819   0.0113823   
thread0::elementwise_mul           1300        38.8029     35.535983 (0.915808)    3.266905 (0.084192)     0.027043    0.054723    0.0298484   0.0103695   
thread0::sequence_softmax          1300        37.2402     33.179899 (0.890971)    4.060267 (0.109029)     0.025292    0.054197    0.0286463   0.00995187  
thread0::softmax                   1300        33.4983     29.525632 (0.881407)    3.972659 (0.118593)     0.022953    0.043861    0.0257679   0.00895191  
thread0::top_k                     1300        32.2755     27.547894 (0.853522)    4.727653 (0.146478)     0.022233    0.042359    0.0248273   0.00862515  
thread0::lookup_table              1300        32.1727     29.858928 (0.928082)    2.313810 (0.071918)     0.0221      0.042696    0.0247483   0.00859768  
thread0::batch_norm                800         29.879      25.255527 (0.845259)    4.623505 (0.154741)     0.026135    0.075775    0.0373488   0.00798472  
thread0::tanh                      1300        26.1229     23.697647 (0.907161)    2.425210 (0.092839)     0.017787    0.035853    0.0200945   0.00698094  
thread0::log                       1300        24.6454     22.828661 (0.926286)    1.816701 (0.073714)     0.016688    0.039488    0.018958    0.0065861   
thread0::relu                      900         16.6115     14.767226 (0.888974)    1.844317 (0.111026)     0.014439    0.042861    0.0184573   0.00443918  
thread0::less_than                 1400        10.563      10.554605 (0.999207)    0.008379 (0.000793)     0.006058    0.022998    0.00754499  0.0028228   
thread0::pool2d                    300         9.70418     8.396219 (0.865217)     1.307956 (0.134783)     0.023248    0.057122    0.0323473   0.00259329  
thread0::concat                    200         9.45507     8.519445 (0.901045)     0.935623 (0.098955)     0.036116    0.06549     0.0472753   0.00252672  
thread0::tensor_array_to_tensor    100         9.39223     9.294675 (0.989613)     0.097559 (0.010387)     0.076668    0.116705    0.0939223   0.00250993  
thread0::scale                     200         8.92703     8.391879 (0.940053)     0.535152 (0.059947)     0.015265    0.094738    0.0446352   0.00238561  
thread0::fetch                     200         7.09898     6.643396 (0.935824)     0.455587 (0.064176)     0.030312    0.052051    0.0354949   0.0018971   
thread0::increment                 1300        6.96634     6.959823 (0.999065)     0.006513 (0.000935)     0.004677    0.020475    0.00535872  0.00186165  
thread0::lod_reset                 1300        6.60527     6.599151 (0.999073)     0.006122 (0.000927)     0.004321    0.01748     0.00508098  0.00176516  
thread0::logical_not               1300        6.04469     6.038321 (0.998947)     0.006364 (0.001053)     0.003846    0.018651    0.00464976  0.00161535  
thread0::logical_and               1300        5.65812     5.653227 (0.999135)     0.004892 (0.000865)     0.003749    0.02097     0.0043524   0.00151205  
thread0::cast                      200         4.24322     4.049011 (0.954231)     0.194206 (0.045769)     0.005938    0.048223    0.0212161   0.00113394  
thread0::im2sequence               100         3.29164     2.681419 (0.814614)     0.610224 (0.185386)     0.026593    0.04697     0.0329164   0.000879642 
thread0::fill_constant             200         1.30061     1.290118 (0.991936)     0.010488 (0.008064)     0.00251     0.019055    0.00650303  0.000347567 
thread0::feed                      300         0.935728    0.924152 (0.987629)     0.011576 (0.012371)     0.001023    0.014889    0.00311909  0.000250059
```
